### DISCUSSION TITLE
How to use  a domain or real IP to access it?

### DISCUSSION THREAD

**Discussion by dexterhuh** [oobabooga/text-generation-webui#1296] [Q&A]
I found --share is similar but seems it creates a gradio.live domain for me for public access. How to use my domain or real IP for public access?

**Comment by BetaDoggo** [comment]
You could use the `--listen` arg and forward the webui port in your router but that would be insecure. A better alternative would be to use a local vpn or an alternative like tailscale or zerotier to connect to the computer running the webui.

Note that sharing your webui instance publicly is a bad idea for security reasons.

**Reply by dexterhuh** [reply]
Thank you very much. It is a very good point for security concerns. Is there a way to do authentication of access? Even manage multiple user access?

**Reply by BetaDoggo** [reply]
There is the `--gradio-auth-path` argument which lets you specify a file with usernames and passwords but since gradio doesn't use https out of the box these can be easily stolen if the instance is used on a public network.

It's all down to your risk tolerance. In theory the worst case would be someone unauthorized using your compute to run the ui, but there are always risks of possible code execution exploits and other such issues.
